File: samples/src/commonMain/kotlin/sample/_class/inherit/Inherit.kt
package sample._class.inherit

import deezer.kustomexport.KustomExport

@KustomExport
interface BaseInterface {
    val baseValue: Long
}

@KustomExport
class BaseImpl : BaseInterface {
    override val baseValue: Long = 42L
}

// Inherit by delegation
@KustomExport
class WrapperImpl(delegate: BaseInterface) : BaseInterface by delegate {
    val wrapperStuff: String = "wrapper"
}

File: samples/src/commonMain/kotlin/sample/_class/inherit/Inherit.ts
import { runTest } from "../../shared_ts/RunTest"
import { assert } from "../../shared_ts/Assert"
import { sample } from '@kustom/Samples'

runTest("Inherit", () : void => {
    var baseImpl = new sample._class.inherit.js.BaseImpl()
    var wrapper = new sample._class.inherit.js.WrapperImpl(baseImpl)
    assert(true, "can instantiate")
})